trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
+        except Exception as exc:
+            raise RuntimeError('issue29692:Chained') from exc
+    try:
+        async with test_issue29692():
+            raise ZeroDivisionError
+    except Exception as ex:
+        assert type(ex) is RuntimeError
+        assert ex.args[0] == 'issue29692:Chained'
+        assert isinstance(ex.__cause__, ZeroDivisionError)
```

Comment:
The problem with writing tests this way is that if no exception is raised, then everything looks good... so it's generally better practice to do:
```python
with pytest.raises(RuntimeError) as excinfo:
    ...
assert excinfo.value.args[0] == "..."
...
```
which will actually fail if it doesn't see a `RuntimeError.` OTOH if this is copied directly from the upstream fix and you'd rather keep it close to upstream there's a reasonable argument for that.